### PR TITLE
feat: add QA for Nano Contract

### PR DESCRIPTION
**Pull request by alexruzenhack**
### Acceptance Criteria
- Add QA document for Nano Contract feature

### Security Checklist
- [x] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: QA_NANO_CONTRACT.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+	- If a failure feedback modal appear with the message "The informed address does not belong to the wallet", it happens when you are using the wallet service. Make sure to disable it and reload the wallet
+- [ ] Check the feedback modal has:
+	- A message informing the registration has happen with success, and
+	- There is a "SEE CONTRACT" CTA
+- Tap on "SEE CONTRACT" CTA
+- [ ] Check *Nano Contract Details* screen is in focus
+
+### Inspect Nano Contract Details
+- From the *Nano Contract Details* screen follow the instructions
+- [ ] Check there is at leas one transaction listed, the "initialize"
```
Comment (pedroferreira1) [inline QA_NANO_CONTRACT.md:78]:
```suggestion
- [ ] Check there is at least one transaction listed, the "initialize"
```

**Review by pedroferreira1 — APPROVED** [review]

File: QA_NANO_CONTRACT.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+	- a message, and
+	- a call-to-action (CTA) "Register new"
+
+### Register Nano Contract
+- Register a new Nano Contract by tapping over "Register new" on *Nano Contract Details*
+	- It may focus on a component of loading
+- [ ] Check the Nano Contract Registration screen is in focus
+- [ ] Check the Nano Contract ID is an input
+- [ ] Check the Wallet Address is loaded
+- [ ] Check the "REGISTER NANO CONTRACT" button is disabled
```
Comment (andreabadesso) [inline QA_NANO_CONTRACT.md:27]:
```suggestion
- [ ] Check if the Nano Contract Registration screen is in focus
- [ ] Check if the Nano Contract ID is an input
- [ ] Check if the Wallet Address is loaded
- [ ] Check if the "REGISTER NANO CONTRACT" button is disabled
```

File: QA_NANO_CONTRACT.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+- Type the following Nano Contract ID `00004f1a3b07f0ffc7ced77a19ae6d4407991a15a281c0ddaf3d6e04a56eba78`
+	- It represents a Nano Contract in the `nano-testnet` network
+- [ ] Check the "REGISTER NANO CONTRACT" button is now actionable
+- Tap on "REGISTER NANO CONTRACT" button
+- [ ] Check for a failure feedback modal to appear
+	- You can't register a Nano Contract from another network
+	- If the message "The informed address does not belong to the wallet" is the failure message, it happens when you are using the wallet service. Make sure to disable it and reload the wallet.
+- Dismiss the modal by tapping outside its area
+- Remove the input
+- [ ] Check a validation message of error appear under the input:
```
Comment (andreabadesso) [inline QA_NANO_CONTRACT.md:37]:
```suggestion
- [ ] Check if a validation message of error appeared under the input:
```

File: QA_NANO_CONTRACT.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+	- It should informe the field is required
+	- it should be in color red
+- [ ] Check the "REGISTER NANO CONTRACT" button is disabled
```
Comment (andreabadesso) [inline QA_NANO_CONTRACT.md:40]:
```suggestion
- [ ] Check if the "REGISTER NANO CONTRACT" button is disabled
```

File: QA_NANO_CONTRACT.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+- Type any input
+- [ ] Check the validation error message disappear
```
Comment (andreabadesso) [inline QA_NANO_CONTRACT.md:42]:
```suggestion
- [ ] Check if the validation error message disappeared
```

File: QA_NANO_CONTRACT.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+### Change network
+- Navigate to the *Settings* screen
+- Go to "Network Settings"
+- Understand disclaimer
+- Customize
```
Comment (andreabadesso) [inline QA_NANO_CONTRACT.md:48]:
```suggestion
- Go to "Network Settings" > Understand disclaimer > Customize
```

File: QA_NANO_CONTRACT.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+- [ ] Check you not see the fields:
```
Comment (andreabadesso) [inline QA_NANO_CONTRACT.md:49]:
```suggestion
- [ ] Check if the following fields are hidden, you should not be able to see them:
```

File: QA_NANO_CONTRACT.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+	- "Wallet Service URL (optional)", and
+	- "Wallet Service WS URL (optional)"
+- If you see the mentioned fields check they are empty or do it by your own before following the next instructions
```
Comment (andreabadesso) [inline QA_NANO_CONTRACT.md:52]: This instruction should never happen as the fields should be hidden

File: QA_NANO_CONTRACT.md [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+### Register Nano Contract using the right network
+- Select the option "Nano Contracts" of the toggle button
+- Tap on "Register new" CTA
+- Type `00004f1a3b07f0ffc7ced77a19ae6d4407991a15a281c0ddaf3d6e04a56eba78` as input to Nano Contract ID
+- Tap on "REGISTER NANO CONTRACT" button
+- [ ] Check for a success feedback modal to appear
+	- If a failure feedback modal appear with the message "The informed address does not belong to the wallet", it happens when you are using the wallet service. Make sure to disable it and reload the wallet
+- [ ] Check the feedback modal has:
+	- A message informing the registration has happen with success, and
+	- There is a "SEE CONTRACT" CTA
```
Comment (andreabadesso) [inline QA_NANO_CONTRACT.md:72]:
```suggestion
	- A "SEE CONTRACT" button
```

File: QA_NANO_CONTRACT.md [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+- Tap on "SEE CONTRACT" CTA
```
Comment (andreabadesso) [inline QA_NANO_CONTRACT.md:73]:
```suggestion
- Tap on "SEE CONTRACT" button
```

File: QA_NANO_CONTRACT.md [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+- Tap on "Registered Address"
+- [ ] Check for a "Choose New Wallet Address" modal appears
+- [ ] Check the address of *index 0* is selected
+- Select any other indexed address
+- [ ] Check for a "New Nano Contract Address" modal appears
+- [ ] Check "New Nano Contract Address" modal has:
+	- An information content informing user this address is used to sign a transaction within the Nano Contract
+	- A mirror content informing the selected address and its index
+	- A "CONFIRM NEW ADDRESS" button
+	- A "GO BAKC" button
```
Comment (andreabadesso) [inline QA_NANO_CONTRACT.md:101]:
```suggestion
	- A "GO BACK" button
```

**Review by andreabadesso — CHANGES_REQUESTED** [review]

File: QA_NANO_CONTRACT.md (hunk 8 above)
Comment (alexruzenhack) [reply QA_NANO_CONTRACT.md:52]: This can happen if Wallet Service is enabled. But I will add a `[NOTE]` prefix to the instruction.

**Review by andreabadesso — APPROVED** [review]

**Pull request merged**